### ISSUE TITLE
Anonymous execution of requests is allowed if the jwt-secret is set.

### ISSUE THREAD

**Issue by SergiyShest** [PostgREST/postgrest#2104]
### Environment

* PostgreSQL version: 13
* PostgREST version: 9
* Operating system:Windows 10

### Description of issue
 
I expected that if I set the jwt-secret parameter in the configuration, it would automatically set that
only jwt-authorization would work.
However, I can make a request without authorization even just typing in the browser:
for example  `http://localhost:3000/myTable ?Id=eq.2.`
I get json with the required data.

At the same time when I make a request with the authorization parameter 
for example: `curl http://localhost:3000/myTable  -H "Authorization: Bearer eyJhbGciOiJIUzI1Ni....`
then the request passes only if the token is correct.

**Comment by wolfgangwalther** [comment]
You provided a lot more context in https://github.com/PostgREST/postgrest-docs/issues/482, specifically:

```
db-uri = "postgres://remote_worker:1HyiYai@localhost:5432/myDb"
db-anon-role = "remote_worker"
```

The first thing that's wrong here is, that you are using the same role as authenticator and as anonymous role. You should use a separate role for anonymous and in your case, this role should have **no privileges at all**, which effectively disables anonymous access. Disabling anonymous access before hitting the database is not implemented, yet, and discussed in #1689.

The second problem is, that your authenticator role (`remote_worker`) has privileges on their own. This role should not have any privileges other than the ability to do `SET ROLE` do a role specified in the JWT. So even if you happen to use it as the anonymous role accidently, those requests should still fail if your privileges were set up correctly.

Please read through https://postgrest.org/en/stable/auth.html#overview-of-role-system to get a better understanding of the role system we are using with PostgREST.

**Comment by SergiyShest** [comment]
Well, thank you. For my purposes, this will do. But still, the current behavior is more like a bug. And what you are suggesting is a way to get around this bug.
26.12.2021, 12:48, Wolfgang Walther < ***@***.*** ***@***.*** >
You provided a lot more context in https://github.com/PostgREST/postgrest-docs/issues/482 PostgREST/postgrest-docs#482, specifically:
db-uri = ***@***.***:5432/myDb"
db-anon-role = "remote_worker" The first thing that's wrong here is, that you are using the same role as authenticator and as anonymous role. You should use a separate role for anonymous and in your case, this role should have no privileges at all, which effectively disables anonymous access. Disabling anonymous access before hitting the database is not implemented, yet, and discussed in https://github.com/PostgREST/postgrest/issues/1689 #1689.
The second problem is, that your authenticator role ( remote_worker ) has privileges on their own. This role should not have any privileges other than the ability to do SET ROLE do a role specified in the JWT. So even if you happen to use it as the anonymous role accidently, those requests should still fail if your privileges were set up correctly.
Please read through https://postgrest.org/en/stable/auth.html#overview-of-role-system https://postgrest.org/en/stable/auth.html#overview-of-role-system to get a better understanding of the role system we are using with PostgREST.
